### PR TITLE
support W3C DID

### PR DESCRIPTION
**Pull request by olomix**

### CODE REVIEW

File: id.go [hunk 1]
```diff
@@ -15,10 +15,6 @@ var (
 	// - first 2 bytes: `00000000 00000000`
 	TypeDefault = [2]byte{0x00, 0x00}
 
-	// TypeReadOnly specifies the readonly identity, this type of identity MUST not be published on chain
-	// - first 2 bytes: `00000000 00000001`
-	TypeReadOnly = [2]byte{0b00000000, 0b00000001}
```
Comment (vmidyllic) [inline id.go:20]: @demonsh should TypeReadOnly be removed?

File: id.go [hunk 2]
```diff
@@ -240,3 +233,9 @@ func CheckGenesisStateID(id, state *big.Int) (bool, error) {
 
 	return id.Cmp(identifier.BigInt()) == 0, nil
 }
+
+func GenesisFromEthAddress(address [20]byte) *big.Int {
```
Comment (vmidyllic) [inline id.go:237]: move function to did.go and we can name it NewDIDFromEthAddress

**Review by demonsh — APPROVED** [review]

**Review by vmidyllic — DISMISSED** [review]

File: w3c/did_w3c.go [hunk 3]
```diff
@@ -0,0 +1,818 @@
+// Package w3c is a set of tools to work with Decentralized Identifiers (DIDs)
+// as described in the DID spec https://w3c.github.io/did-core/
+// Got from https://github.com/build-trust/did
```
Comment (OBrezhniev) [inline w3c/did_w3c.go:3]: Files in this package should have copyright notice with license and original authors.

**Comment by ilya-korotya** [comment]
The `EthAddressFromID` function is inconvenient because it returns the Ethereum address as an array of bytes. What is the correct way to convert these bytes into a string representation (e.g., `string(contractAddress[:])`)? If EthAddressFromID could directly return the string representation, it would be much more convenient.

**Comment by ilya-korotya** [comment]
```go
did, err := w3c.ParseDID(issuer)
if err != nil {
  return verifiable.CredentialStatus{}, err
}
id, err := core.IDFromString(did.ID)
if err != nil {
  return verifiable.CredentialStatus{}, err
}
contractAddress, err := core.EthAddressFromID(id)
```
Two points for this snippet:
1. Inconsistency in naming: You used ParseDID to obtain a DID from a string, but the ID package uses IDFromString for the same functionality.
2. Since all functions that process ID work with ID objects, not strings, can we refactor the DID structure to accommodate this?
```go
type DID struct {
   ID core.ID // not string
```

**Review by vmidyllic — APPROVED** [review]

**Review by vmidyllic — APPROVED** [review]

**Review by demonsh — APPROVED** [review]

**Pull request merged**